front/app/modules/commercial/insights/api/views/useUpdateView.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import viewsKeys from './keys';
+import { IInsightsView } from './types';
+
+interface IInsightViewUpdateObject {
+  id: string;
+  requestBody: { view: { name: string } };
+}
+
+const updateView = ({ id, requestBody }: IInsightViewUpdateObject) =>
+  fetcher<IInsightsView>({
+    path: `/insights/views/${id}`,
+    action: 'patch',
+    body: requestBody,
+  });
+
+const useUpdateView = ({ onSuccess }: { onSuccess?: () => void }) => {
+  const queryClient = useQueryClient();
+  return useMutation<IInsightsView, CLErrors, IInsightViewUpdateObject>({
+    mutationFn: updateView,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: viewsKeys.lists() });
+      onSuccess && onSuccess();
```

Comment:
We have a new way to handle these types of callbacks, it is possible to pass them directly to your mutation like so:
```
  const handleSubmit = async () => {
    if (name) {
      mutate(
        { id: insightsViewId, requestBody: { view: { name } } },
        { onSuccess: closeRenameModal }
      );
    }
  };
```

It is much better because it provides a ton more flexibility. The response data is also available to the onSuccess callback out of the box -> `onSuccess: (data) => // Do something with the response data in your component, if needed`